api4/webhook.go

```diff
@@ -83,3 +87,79 @@ func getIncomingHooks(c *Context, w http.ResponseWriter, r *http.Request) {
 
 	w.Write([]byte(model.IncomingWebhookListToJson(hooks)))
 }
+
+func getIncomingHook(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireHookId()
+	if c.Err != nil {
+		return
+	}
+
+	hookID := c.Params.HookId
+
+	var err *model.AppError
+	var hook *model.IncomingWebhook
+	var channel *model.Channel
+
+	if hook, err = app.GetIncomingWebhook(hookID); err != nil {
+		c.Err = err
+		return
+	} else {
+		channel, err = app.GetChannel(hook.ChannelId)
+		if err != nil {
+			c.Err = err
+			return
+		}
+
+		fmt.Println("session has permission to team ", app.SessionHasPermissionToTeam(c.Session, hook.TeamId, model.PERMISSION_MANAGE_WEBHOOKS))
+		fmt.Println("Channel open ", channel.Type)
+		fmt.Println("session has permission to channel ", app.SessionHasPermissionToChannel(c.Session, hook.ChannelId, model.PERMISSION_MANAGE_WEBHOOKS))
+
+		if !app.SessionHasPermissionToTeam(c.Session, hook.TeamId, model.PERMISSION_MANAGE_WEBHOOKS) ||
+			(channel.Type != model.CHANNEL_OPEN && !app.SessionHasPermissionToChannel(c.Session, hook.ChannelId, model.PERMISSION_MANAGE_WEBHOOKS)) {
```

Comment:
I think the permission check for the channel bit should be for `model.PERMISSION_READ_CHANNEL`. The team permission check will already check if they have permission to manage webhooks, then you just need to check if they are in the channel if it's not open with the second one

Does that make sense?